tests/HTMLPurifierTest.hack

```diff
@@ -466,4 +466,83 @@ class HTMLPurifierTest extends HackTest {
 		expect($clean_html)->toEqual($expected_html);
 		echo "finished.\n\n";
 	}
+
+	public function testRel(): void {
+		echo "\nrunning testRel()...";
+		$policy = new HTMLPurifier\HTMLPurifier_Policy(
+			dict[
+				'b' => vec[],
+				'ul' => vec[],
+				'li' => vec[],
+				'ol' => vec[],
+				'h2' => vec[],
+				'h4' => vec[],
+				'br' => vec[],
+				'div' => vec[],
+				'strong' => vec[],
+				'del' => vec[],
+				'em' => vec[],
+				'pre' => vec[],
+				'code' => vec[],
+				'table' => vec[],
+				'tbody' => vec[],
+				'td' => vec[],
+				'th' => vec[],
+				'thead' => vec[],
+				'tr' => vec[],
+				'a' => vec['id', 'name', 'href', 'target', 'rel'],
+				'h3' => vec['class'],
+				'p' => vec['class'],
+				'aside' => vec['class'],
+				'img' => vec['src', 'alt', 'class', 'width', 'height', 'srcset', 'sizes'],
+			],
+		);
+		$config = HTMLPurifier\HTMLPurifier_Config::createDefault();
+		$purifier = new HTMLPurifier\HTMLPurifier($config, $policy);
+		$dirty_html = '<p><a rel="ab" name="bobcat"></a></p>';
+		$clean_html = $purifier->purify($dirty_html);
+		$expected_html = '<p><a name="bobcat"></a></p>';
+		expect($clean_html)->toEqual($expected_html);
+		echo "finished.\n\n";
+	}
+
+
+	public function testAtagTargetAttribute(): void {
+		echo "\nrunning testAtagTargetAttribute()...";
+		$policy = new HTMLPurifier\HTMLPurifier_Policy(
+			dict[
+				'b' => vec[],
+				'ul' => vec[],
+				'li' => vec[],
+				'ol' => vec[],
+				'h2' => vec[],
+				'h4' => vec[],
+				'br' => vec[],
+				'div' => vec[],
+				'strong' => vec[],
+				'del' => vec[],
+				'em' => vec[],
+				'pre' => vec[],
+				'code' => vec[],
+				'table' => vec[],
+				'tbody' => vec[],
+				'td' => vec[],
+				'th' => vec[],
+				'thead' => vec[],
+				'tr' => vec[],
+				'a' => vec['id', 'name', 'href', 'target', 'rel'],
+				'h3' => vec['class'],
+				'p' => vec['class'],
+				'aside' => vec['class'],
+				'img' => vec['src', 'alt', 'class', 'width', 'height', 'srcset', 'sizes'],
+			],
+		);
+		$config = HTMLPurifier\HTMLPurifier_Config::createDefault();
+		$purifier = new HTMLPurifier\HTMLPurifier($config, $policy);
+		$dirty_html = '<p><a name="bobcat" target="_blank"></a></p>';
+		$clean_html = $purifier->purify($dirty_html);
+		$expected_html = '<p><a name="bobcat" target="_blank" rel="noopener noreferrer"></a></p>';
```

Comment:
A few more tests I can think of here:

```
Make sure noopener noreferrer is preserved
Dirty: <p><a name="bobcat" target="_blank" rel="noopener noreferrer"></a></p>
Clean: <p><a name="bobcat" target="_blank" rel="noopener noreferrer"></a></p>
```

```
Make sure incorrect rels are stripped and correct ones are added:
Dirty: <p><a name="bobcat" target="_blank" rel="ab" target="_blank"></a></p>
Clean: <p><a name="bobcat" target="_blank" rel="noopener noreferrer"></a></p>
```

```
Make sure unwanted rels are stripped:
Dirty: <p><a name="bobcat" target="_blank" rel="ab noopener noreferrer"></a></p>
Clean: <p><a name="bobcat" target="_blank" rel="noopener noreferrer"></a></p>
```

```
Not sure what behavior this one should have but let's test it?:
Dirty: <p><a name="bobcat" rel="noopener noreferrer"></a></p>
Clean: ?
```

Reply:
I think last one clean is the same as dirty.